src/routes/safe/store/actions/transactions/fetchTransactions/index.ts

```diff
@@ -7,28 +7,39 @@ import { loadOutgoingTransactions } from './loadOutgoingTransactions'
 
 import { addOrUpdateCancellationTransactions } from 'src/routes/safe/store/actions/transactions/addOrUpdateCancellationTransactions'
 import { addOrUpdateTransactions } from 'src/routes/safe/store/actions/transactions/addOrUpdateTransactions'
+import { Dispatch } from 'redux'
+import { backOff } from 'exponential-backoff'
 
 const noFunc = () => {}
 
-export default (safeAddress: string) => async (dispatch) => {
-  const transactions = await loadOutgoingTransactions(safeAddress)
-
-  if (transactions) {
-    const { cancel, outgoing } = transactions
-    const updateCancellationTxs = cancel.size
-      ? addOrUpdateCancellationTransactions({ safeAddress, transactions: cancel })
-      : noFunc
-    const updateOutgoingTxs = outgoing.size ? addOrUpdateTransactions({ safeAddress, transactions: outgoing }) : noFunc
-
-    batch(() => {
-      dispatch(updateCancellationTxs)
-      dispatch(updateOutgoingTxs)
-    })
-  }
-
-  const incomingTransactions = await loadIncomingTransactions(safeAddress)
-
-  if (incomingTransactions.get(safeAddress).size) {
-    dispatch(addIncomingTransactions(incomingTransactions))
+export default (safeAddress: string) => async (dispatch: Dispatch): Promise<void> => {
+  try {
+    const transactions = await backOff(() => loadOutgoingTransactions(safeAddress))
```

Comment:
I would set a maximum delay of like 5 or 10 min else this will completely break the ux if the service is down for a couple minutes.

Reply:
it will be called again

Reply:
it fetches the data continuously, if this request fails, the interface will send a new one

Reply:
Yes, but it increases the delay (*2) so at some point this will be a very long delay (not critical as it can be reset with a reload :) )

Reply:
@rmeissner https://github.com/coveooss/exponential-backoff#readme the defaults seem to be ok imo

Reply:
Actually looking at it the default values are not the best for oir use case. 100ms as a default delay makes not too much sense in my opinion as this will only triggered more failed request really fast. I would probably set this to 5 or 10 seconds. And in this case 10 retries with infinitive delay is quite a lot  (~1.5h delay with 5 seconds)